contracts/SupportsInterfaceWithLookup.sol

```diff
@@ -0,0 +1,52 @@
+pragma solidity ^0.4.23;
+
+import "./ERC165.sol";
+
+
+/**
+ * @title SupportsInterfaceWithLookup
+ * @author Matt Condon (@shrugs)
+ * @dev Implements ERC165 using a lookup table.
+ */
+contract SupportsInterfaceWithLookup is ERC165 {
+  bytes4 public constant InterfaceId_ERC165 = 0x01ffc9a7;
+  /**
+   * 0x01ffc9a7 ===
+   *   bytes4(keccak256('supportsInterface(bytes4)'))
+   */
+
+  /**
+   * @dev a mapping of interface id to whether or not it's supported
+   */
+  mapping(bytes4 => bool) internal supportedInterfaces;
+
+  /**
+   * @dev A contract implementing SupportsInterfaceWithLookup
+   * @dev  implement ERC165 itself
+   */
+  constructor()
+    public
+  {
+    _registerInterface(InterfaceId_ERC165);
+  }
+
+  /**
+   * @dev implement supportsInterface(bytes4) using a lookup table
+   */
+  function supportsInterface(bytes4 _interfaceId)
+    external
+    view
+    returns (bool)
+  {
+    return supportedInterfaces[_interfaceId];
+  }
+
+  /**
+   * @dev private method for registering an interface
+   */
+  function _registerInterface(bytes4 _interfaceId)
```

Comment:
should we check here that _interfaceId is not `0xffffffff`?
And another question, where does that `0xffffffff` comes from?

Reply:
The 0xff... is a magic value. If it returns true for that query, the chances of the contract having implemented the function `supportsInterface(bytes4)` is infinitesimally low (since the return value of a function which has no return value is garbage data, which is, 99.99999999...% of the time != false. so if the function returns true, it wasn't implemented.

Reply:
Interesting. I wonder why this is not explained in the EIP.

Reply:
yeah I had to come to that conclusion on my own; maybe it's an existing pattern? comments don't mention it either ¯\_(ツ)_/¯